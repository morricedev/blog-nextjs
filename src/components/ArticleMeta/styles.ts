import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkGray};
    font-style: italic;

    .categories span::after {
      content: ', ';
    }

    .categories span:last-child::after {
      content: '';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 0.3s ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
