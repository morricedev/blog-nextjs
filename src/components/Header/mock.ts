export default {
  blogName: 'Morrice',
  blogDescription:
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Nulla ad doloribus omnis eaque obcaecati magnam cum sequi ab ducimus nemo officia illo.',
  logo: 'assets/images/teste.jpg',
};
