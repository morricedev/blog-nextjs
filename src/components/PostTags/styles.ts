import styled, { css } from 'styled-components';

export const Wrapper = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;

    span {
      margin: 0 0 0 0.5rem;

      &::after {
        content: ', ';
      }

      &:last-child::after {
        content: '';
      }
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 0.3s ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
