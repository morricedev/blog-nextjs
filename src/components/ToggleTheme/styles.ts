import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  position: fixed;
  top: 2rem;
  right: 2rem;
  z-index: 2;
`;

export const Label = styled.label`
  position: relative;
  display: inline-block;
  width: 5rem;
  height: 2.5rem;
  line-height: 0;
  font-size: 0;
  overflow: hidden;
  color: transparent;
`;

export const Input = styled.input`
  ${({ theme }) => css`
    appearance: none;
    opacity: 0;
    width: 0;
    height: 0;

    &:checked + ${Slider} {
      background: ${theme.colors.secondary};

      &:before {
        transform: translateX(1.9rem);
        background: white;
      }
    }

    &:focus + ${Slider} {
      border: 1px solid blue;
    }
  `}
`;

export const Slider = styled.span`
  ${({ theme }) => css`
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: ${theme.colors.darkGray};
    transition: all 300ms ease-in-out;
    border-radius: 2rem;
    border: 1px solid transparent;

    &:before {
      content: '';
      position: absolute;
      height: 2.1rem;
      width: 2.1rem;
      left: 0.4rem;
      bottom: 0.1rem;
      background: ${theme.colors.primary};
      border-radius: 50%;
      transition: all 300ms ease-in-out;
    }
  `}
`;
