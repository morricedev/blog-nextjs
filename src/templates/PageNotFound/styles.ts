import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;
    height: 80vh;
    margin: 0 auto;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    text-align: center;

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 0.3s ease-in-out;

      &:hover {
        filter: brightness(80%);
      }
    }
  `}
`;
