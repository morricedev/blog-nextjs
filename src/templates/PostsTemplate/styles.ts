import styled, { css } from 'styled-components';

export const ButtonContainer = styled.div`
  ${({ theme }) => css`
    padding: 0 ${theme.spacings.large};
    margin: ${theme.spacings.large} 0;
    display: flex;
    justify-content: center;
  `}
`;

export const SearchContainer = styled.div`
  ${({ theme }) => css`
    margin: 0 auto;
    margin-bottom: ${theme.spacings.xlarge};
    padding: 0 ${theme.spacings.large};
    max-width: 120rem;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
  `}
`;

export const SearchInput = styled.input`
  ${({ theme }) => css`
    padding: 0.5rem ${theme.spacings.small};

    &:disabled {
      outline: 0.1rem solid ${theme.colors.secondary};
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    border: none;
    padding: ${theme.spacings.small} ${theme.spacings.large};
    cursor: pointer;

    &:disabled {
      background: ${theme.colors.darkGray};
    }
  `}
`;
