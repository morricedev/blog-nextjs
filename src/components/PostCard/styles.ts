import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      margin: ${theme.spacings.small} 0;
    }

    a {
      text-decoration: none;
      color: inherit;
      transition: color 0.3s ease-in-out;
    }

    &:hover {
      a {
        color: ${theme.colors.secondary};
      }

      img {
        opacity: 0.8;
      }
    }
  `}
`;

export const Cover = styled.img`
  max-width: 100%;
  height: 20rem;
  transition: opacity 0.3s ease-in-out;
`;

export const Excerpt = styled.p``;
