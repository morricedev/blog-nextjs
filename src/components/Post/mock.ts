import { PostProps } from '.';
import { data } from '../../api/data.json';
import mock from '../HtmlContent/mock';

const { title, excerpt, cover, author, categories, createdAt } = data.posts[0];

export default {
  title,
  excerpt,
  cover,
  content: mock,
  author,
  categories,
  createdAt,
} as PostProps;
