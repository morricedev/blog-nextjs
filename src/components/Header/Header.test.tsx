import { screen } from '@testing-library/react';

import { renderTheme } from '../../styles/render-theme';
import { Header } from '.';
import mock from './mock';

describe('<Header/>', () => {
  it('should render an image, a heading and text', () => {
    renderTheme(<Header {...mock} showText={undefined} />);

    expect(
      screen.getByRole('heading', { name: mock.blogName }),
    ).toBeInTheDocument();

    expect(screen.getByRole('img', { name: /Morrice/i })).toHaveAttribute(
      'src',
      mock.logo,
    );
  });

  it('should render image only', () => {
    renderTheme(<Header {...mock} showText={false} />);

    expect(
      screen.queryByRole('heading', { name: mock.blogName }),
    ).not.toBeInTheDocument();

    expect(screen.getByRole('img', { name: /Morrice/i })).toHaveAttribute(
      'src',
      mock.logo,
    );
  });

  it('should match snapshot', () => {
    const container = renderTheme(<Header {...mock} showText={false} />);
    expect(container).toMatchSnapshot();
  });
});
