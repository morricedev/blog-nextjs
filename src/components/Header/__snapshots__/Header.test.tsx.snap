// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c2 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c3 > img {
  width: 15rem;
  height: 15rem;
  border-radius: 50%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto;
  padding-bottom: 6.4rem;
  width: 100%;
  max-width: 96rem;
  color: #333333;
  font-size: 1.6rem;
}

.c0 .c1 {
  margin: 0 0 calc(1.6rem - 1rem);
}

@media (max-width:360px) {
  .c0 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
  }

  .c0 > .c1 {
    margin: 0 0 2.4rem 0;
  }
}

<body>
    <div>
      <header
        class="c0"
      >
        <h1
          class="c1 c2"
        >
          <a
            class="c3"
            href="/"
            target="_self"
          >
            <img
              alt="Morrice - Blog"
              src="assets/images/teste.jpg"
            />
          </a>
        </h1>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-dlnjwi bDQZOL"
    >
      <h1
        class="sc-bdnxRM htrKD"
      >
        <a
          class="sc-gtsrHT cNievg"
          href="/"
          target="_self"
        >
          <img
            alt="Morrice - Blog"
            src="assets/images/teste.jpg"
          />
        </a>
      </h1>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
